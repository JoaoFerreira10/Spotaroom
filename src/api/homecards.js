
export default class homecards {

    static instance = null;

    static getInstance () {

        if(!this.instance)
            this.instance = new homecards();

        return this.instance;
    }

    async getHomecards () {

        try {

            let data = await fetch('/api/homecards');
            return data;
        } catch (error) {

            throw new Error("ERROR: "+error);
        }
    }
}