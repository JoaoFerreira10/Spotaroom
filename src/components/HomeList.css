.home-list {
    top: 60px;
    position: relative;
}