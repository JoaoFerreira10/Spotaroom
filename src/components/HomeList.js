import React from 'react';
import HomeItem from './HomeItem';
import './HomeList.css';

const HomeList = ({ data }) => {

    if(!data)
        return <div>No data</div>;  
     
    const homeItems = data.map((elem, index) => {

        return <HomeItem key={index} item={elem} />
    });

    return <div className="home-list">{homeItems}</div>;
}

export default HomeList;