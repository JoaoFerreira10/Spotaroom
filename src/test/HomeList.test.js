import React from 'react';
import { create } from 'react-test-renderer';
import HomeList from '../components/HomeList';
import homecards from '../api/homecards'; 

describe("HomeList component test", () => {

  it("it matches with the snapshot", async () => {

    const result = await homecards.getInstance().getHomecards();
    const component = create(<HomeList data={result.homecards}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});