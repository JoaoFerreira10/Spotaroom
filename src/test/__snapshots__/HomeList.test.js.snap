// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeList component test it matches with the snapshot 1`] = `
<div>
  No data
</div>
`;

exports[`MenuBar component test it matches with the snapshot 1`] = `
<div>
  No data
</div>
`;
